octobot/community/feeds/community_mqtt_feed.py

```diff
@@ -0,0 +1,344 @@
+#  This file is part of OctoBot (https://github.com/Drakkar-Software/OctoBot)
+#  Copyright (c) 2023 Drakkar-Software, All rights reserved.
+#
+#  OctoBot is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  OctoBot is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public
+#  License along with OctoBot. If not, see <https://www.gnu.org/licenses/>.
+import logging
+import typing
+import gmqtt
+import json
+import asyncio
+import packaging.version as packaging_version
+
+import octobot_commons.enums as commons_enums
+import octobot_commons.errors as commons_errors
+import octobot_commons.constants as commons_constants
+import octobot.community.errors as errors
+import octobot.community.feeds.abstract_feed as abstract_feed
+import octobot.constants as constants
+
+
+def _disable_gmqtt_info_loggers():
+    logging.getLogger("gmqtt.client").setLevel(logging.WARNING)
+
+
+class CommunityMQTTFeed(abstract_feed.AbstractFeed):
+    MQTT_VERSION = gmqtt.constants.MQTTv311
+    MQTT_BROKER_PORT = 1883
+    RECONNECT_DELAY = 15
+    RECONNECT_ENSURE_DELAY = 1
+    MAX_MESSAGE_ID_CACHE_SIZE = 100
+    MAX_SUBSCRIPTION_ATTEMPTS = 5
+    DISABLE_RECONNECT_VALUE = -2
+    DEVICE_CREATE_TIMEOUT = 5 * commons_constants.MINUTE_TO_SECONDS
+    DEVICE_CREATION_REFRESH_DELAY = 2
+    CONNECTION_TIMEOUT = 10
+
+    # Quality of Service level determines the reliability of the data flow between a client and a message broker.
+    # The message may be sent in three ways:
+    # QoS 0: the message will be received at most once (also known as “fire and forget”).
+    # QoS 1: the message will be received at least once.
+    # QoS 2: the message will be received exactly once.
```

Comment:
I think we can use this one as we use the "best" hub plan

Reply:
we already have an internal check to avoid processing twice the same messages. QoS 2 doubles the number of transmitted messages (ack system), I'm not sure we need it

Reply:
Ok 👍